platform-dom/src/main/java/com/softicar/platform/dom/elements/image/viewer/DomImageViewerToolBar.java

```diff
@@ -15,13 +17,25 @@ public DomImageViewerToolBar(DomImageViewer viewer) {
 
 		addCssClass(DomCssClasses.DOM_IMAGE_VIEWER_TOOL_BAR);
 		appendChild(navigationBar);
+		appendChild(new DomImageViewerToolBarSpacer());
 		appendChild(zoomBar);
-		appendChild(new DomImageViewerRotationButton(viewer));
+		appendChild(new DomImageViewerToolBarSpacer());
+		appendChild(new DomImageViewerTagToggleButton(viewer));
+		appendChild(new DomImageViewerRotateLeftButton(viewer));
+		appendChild(new DomImageViewerRotateRightButton(viewer));
 	}
 
 	public void refresh() {
 
 		zoomBar.refresh();
 		navigationBar.refresh();
 	}
+
+	private static class DomImageViewerToolBarSpacer extends DomDiv {
```

Comment:
hmm, I am okay with this, but doesn't such a spacer run against the idea of CSS? alternatively, one could group the buttons into some `DomDiv` or `DomSpan` or `DomBar` and handle the spacing using pure CSS. Would that be better?

Reply:
I don't know, so let it be.